addons/spreadsheet/static/src/assets_backend/pivot/pivot_model.js

```diff
@@ -0,0 +1,110 @@
+/** @odoo-module **/
+
+import { patch } from "@web/core/utils/patch";
+import { PivotModel } from "@web/views/pivot/pivot_model";
+import { isDateField, normalizeServerValue, parseGroupField } from "./pivot_utils";
+
+patch(PivotModel.prototype, {
```

Comment:
would there be a way to patch the model (or even change the original model) such that the necessary data (from read_group result) is kept in the data structure(s) in such a way that we could create the mapping afterwards (when inserting to the spreadsheet) ?
The goal would be to have a less invasive patch (and less code moved in the main backend assets bundle)